runtime/mangata-kusama/src/lib.rs

```diff
@@ -793,16 +790,20 @@ impl_runtime_apis! {
 			System::set_block_seed(&seed);
 		}
 
-
 		fn get_previous_block_txs() -> Vec<Vec<u8>> {
 			System::get_previous_blocks_txs()
 		}
 
-		fn pop_txs(count: u64) -> Vec<Vec<u8>>{
+		fn pop_txs(count: u64) -> Vec<Vec<u8>> {
 			System::pop_txs(count as usize)
 		}
 
-		fn create_enqueue_txs_inherent(txs: Vec<<Block as BlockT>::Extrinsic>) -> <Block as BlockT>::Extrinsic{
+		fn create_enqueue_txs_inherent(txs: Vec<<Block as BlockT>::Extrinsic>) -> <Block as BlockT>::Extrinsic {
+			for t in txs.iter() {
+				if let Some((_, _, extra)) = &t.signature {
+					let _ = extra.additional_signed();
+				}
+			}
```

Comment:
this is important, a bit hackish

during block building, state access is recorded and then passed to relay to validate block
previously these 'next block' txs were validated in block builder within a transaction that was rolled back eventually, all was well
in validate function the relay validated the trxs correctly

now the rollback also reverts proof recorder
the relay in validate function now fails because of access to txs extras, specifically the mortality check with block number storage access
the storage does not exist in the supplied recorded proof

this API is called during last stages of block building, we evaluate the extras so that the storage access is recored in proof and the validation fucntion on relay succeeds